File: problem-17/main.py
ones = {
        1:'one',
        2:'two',
        3:'three',
        4:'four',
        5:'five',
        6:'six',
        7:'seven',
        8:'eight',
        9:'nine',
}

teens = {
        10:'ten',
        11:'eleven',
        12:'twelve',
        13:'thirteen',
        14:'fourteen',
        15:'fifteen',
        16:'sixteen',
        17:'seventeen',
        18:'eighteen',
        19:'nineteen',
}

tens = {
        20:'twenty',
        30: 'thirty',
        40: 'forty',
        50: 'fifty',
        60: 'sixty',
        70: 'seventy',
        80: 'eighty',
        90: 'ninety',
}
def number_to_words(number):
    if number == 1000:
        return 'onethousand'
    
    word = ''
    hundreds = number // 100
    remainder = number % 100
    
    if hundreds > 0:
        word += ones[hundreds] + 'hundred'
        if remainder > 0:
            word += 'and'
    
    if remainder > 0:
        if remainder < 10:
            word += ones[remainder]
        elif 10 <= remainder < 20:
            word += teens[remainder]
        else:
            tens_part = (remainder // 10) * 10
            ones_part = remainder % 10
            word += tens[tens_part]
            if ones_part > 0:
                word += ones[ones_part]
    
    return word

total_letters = 0

for number in range(1, 1001):
    word = number_to_words(number)
    total_letters += len(word)

print(total_letters)

File: problem-17/main.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char* number_to_words(int number) {
    const char* ones[] = {"", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
    const char* teens[] = {"ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"};
    const char* tens[] = {"", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"};

    if(number==1000) {
        char* word=(char*)malloc(strlen("onethousand")+1);
        strcpy(word,"onethousand");
        return word;
    }

    char* word=(char*)malloc(100);
    word[0]='\0';
    int hundreds=number/100;
    int remainder=number%100;

    if(hundreds>0) {
        strcat(word,ones[hundreds]);
        strcat(word,"hundred");
        if(remainder>0){
            strcat(word,"and");
        }
    }
    if (remainder > 0) {
        if (remainder < 10) {
            strcat(word, ones[remainder]);
        } else if (remainder >= 10 && remainder < 20) {
            strcat(word, teens[remainder - 10]);
        } else {
            int tens_part = remainder / 10;
            int ones_part = remainder % 10;
            strcat(word, tens[tens_part]);
            if (ones_part > 0) {
                strcat(word, ones[ones_part]);
            }
        }
    }

    return word;
}
int main() {
    int total_letters=0;
    for(int number=1;number<=1000;number++) {
        char* word=number_to_words(number);
        total_letters+=strlen(word);
        free(word);
    }
    printf("%d\n",total_letters);
    return 0;

}

